src/streamlink/plugins/twitch.py

```diff
@@ -697,12 +703,19 @@ def _check_for_host(self):
             self.logger.info("{0} is hosting {1}".format(self.channel, host_info["target_login"]))
             return host_info["target_login"]
 
+    def _check_for_rerun(self):
+        return self.api.streams(self.channel_id)["stream"]["stream_type"] == "rerun"
```

Comment:
Btw, according to the [kraken v5 documentation](https://dev.twitch.tv/docs/v5/reference/streams#get-live-streams), supported values are `live`, `playlist` and `all`, but that's not true. There's also a second attribute, `broadcast_platform`, which isn't documented, but is related to reruns and IIRC, it was used by Twitch before they added the `stream_type` attribute. Both attributes have changed quite a bit over the past years after reruns were implemented by Twitch. For example, there also was the `watch_party` value which was used to indicate reruns.

Even if `stream_type` is the documented property, I am not sure if it's the right property to check. Most attribute names can be used for filtering streams via a GET parameter, like `game` for example, but `stream_type` can not be used for filtering whereas `broadcast_platform` can, which is weird. See here:
```bash
curl -s \
  -H "Accept: application/vnd.twitchtv.v5+json" \
  -H "Client-ID: phiay4sq36lfv9zu7cbqwz2ndnesfd8" \
  "https://api.twitch.tv/kraken/streams?stream_type=rerun&limit=1" | \
  jq '.streams[0] | {stream_type: .stream_type, broadcast_platform: .broadcast_platform}'
```
```json
{
  "stream_type": null,
  "broadcast_platform": null
}
```
```bash
curl -s \
  -H "Accept: application/vnd.twitchtv.v5+json" \
  -H "Client-ID: phiay4sq36lfv9zu7cbqwz2ndnesfd8" \
  "https://api.twitch.tv/kraken/streams?broadcast_platform=rerun&limit=1" | \
  jq '.streams[0] | {stream_type: .stream_type, broadcast_platform: .broadcast_platform}'
```
```json
{
  "stream_type": "rerun",
  "broadcast_platform": "rerun"
}
```

Then there's also the problem of broadcasters not setting the rerun flag while streaming a rerun. The Twitch GUI does some additional regex checks of the channel's status message (stream title) in addition to checking both `stream_type` and `broadcast_platform` attributes.

Reply:
>Btw, according to the kraken v5 documentation, supported values are live, playlist and all

To be honest I totally forgot to check the kraken documentation. I was basically going by what the website was doing. Wasn't the playlist feature replaced by the reruns feature anyway? Maybe they are just no longer updating the docs?

>stream_type can not be used for filtering whereas broadcast_platform can, which is weird

Filtering by `stream_type` was removed on their Helix API a while ago. Maybe the same issues / reasons existed on Kraken too and they decided to remove it there as well?

>Then there's also the problem of broadcasters not setting the rerun flag while streaming a rerun.

My intention was to only disable reruns done via Twitch's rerun feature. Are you sure broadcasters have to flag their stream as a rerun when using it?

Reply:
> I totally forgot to check the kraken documentation

The documentation is complete garbage, so forget about that. Checking for `rerun` as value should be correct, but what I meant is that I'm not sure if it's enough since there's also the `broadcast_platform` attribute. The entire API is a total mess and you can't be 100% sure unless they document it properly, which they don't. Not sure about the playlist streams and whether they were removed, but there's also the `is_playlist` attribute. I think they added this after the playlist stream announcement and the documentation is just old and therefore still includes the `playlist` value.

> Filtering by `stream_type` was removed on their Helix API a while ago.

I haven't really followed the past developments on the helix API, but if you say that `stream_type` has been there and was removed (it's currently not there), then it's probably the right attribute to check on kraken and they've simply forgot to remove it there as well. The fact that it's been removed doesn't sound good though.

> Are you sure broadcasters have to flag their stream as a rerun when using it?

What I meant is that there are many channels which don't set the rerun flag while re-streaming a locally recorded stream and they just set a rerun stream title. See the various ESL channels for example (Dota2, CSGO, etc).